ocrd_models/ocrd_models/ocrd_mets.py

```diff
@@ -50,20 +51,91 @@ def empty_mets(now=None):
         tpl = METS_XML_EMPTY.decode('utf-8')
         tpl = tpl.replace('{{ VERSION }}', VERSION)
         tpl = tpl.replace('{{ NOW }}', '%s' % now)
-        return OcrdMets(content=tpl.encode('utf-8'))
+        return OcrdMets(content=tpl.encode('utf-8'), cache_flag=cache_flag)
 
     def __init__(self, **kwargs):
         """
 
         """
         super(OcrdMets, self).__init__(**kwargs)
 
+        # If cache is enabled
+        if self._cache_flag:
+            # Cache for the fileGrps (mets:fileGrp) - a dictionary with Key and Value pair:
+            # Key: 'fileGrp.USE'
+            # Value: a 'fileGrp' object at some memory location 
+            self._fileGrp_cache = {}
+
+            # Cache for the files (mets:file) - two nested dictionaries
+            # The outer dictionary's Key: 'fileGrp.USE'
+            # The outer dictionary's Value: Inner dictionary
+            # The inner dictionary's Key: 'file.ID'
+            # The inner dictionary's Value: a 'file' object at some memory location
+            self._file_cache = {}
+
+            # Note, if the empty_mets() function is used to instantiate OcrdMets
+            # Then the cache is empty even after this operation
+            self._fill_caches()
+
+    def __exit__(self):
+        if self._cache_flag:
+            self._clear_caches()
+
     def __str__(self):
         """
         String representation
         """
         return 'OcrdMets[fileGrps=%s,files=%s]' % (self.file_groups, list(self.find_files()))
 
+    def _fill_caches(self):
+        """
+        Fills the caches with fileGrps and FileIDs
+        """
+
+        tree_root = self._tree.getroot()
+        el_fileGrp_list = tree_root.find(".//mets:fileSec", NS)
+        if el_fileGrp_list is None or len(el_fileGrp_list) == 0:
+            return
+
+        for el_fileGrp in el_fileGrp_list:
+            fileGrp_use = el_fileGrp.get('USE')
+
+            # NOTE: For some reason the el_fileGrp_list contains None values
+            # when testing with the SBB0000F29300010000/data/mets.xml
+            if fileGrp_use is None:
+                continue
```

Comment:
That happens because that METS contains comments which lxml considers to be elements.

```suggestion
            if el_fileGrp.tag is ET.Comment:
                continue
            fileGrp_use = el_fileGrp.get('USE')
```

Reply:
Okay.